backend/tests/integrated/extract/adapters/test_repositories.py

```diff
@@ -0,0 +1,116 @@
+from django.db import DatabaseError
+from pytest_mock import MockFixture
+
+from infrastructures.apps.common import exceptions as common_exceptions
+from infrastructures.apps.extract import exceptions, models
+from modules.extract.domain import ports, value_objects
+from tests import test_const
+
+
+def test_django_file_domain_repository_save_method_saves_file_from_bytes(
+    tmp_path,
+    test_django_file_domain_repository: ports.AbstractFileDomainRepository,
+):
+    # Given
+    with open(test_const.CORRECT_INPUT_CSV, "rb") as file:
+        file_bytes = file.read()
+
+    # When
+    result = test_django_file_domain_repository.save(
+        file_name=os.path.basename(test_const.CORRECT_INPUT_CSV),
+        file=file_bytes,
+        location=str(tmp_path),
+    )
+
+    # Then
+    assert isinstance(result, str)
+    assert (tmp_path / result).exists()
+
+
+def test_django_file_domain_repository_save_method_raises_file_save_error_on_os_error(
+    tmp_path,
+    mocker: MockFixture,
+    test_django_file_domain_repository: ports.AbstractFileDomainRepository,
+):
+    # Given
+    with open(test_const.INCORRECT_INPUT, "rb") as file:
+        file_bytes = file.read()
+
+    side_effect = OSError
+    mocker.patch.object(FileSystemStorage, "save", side_effect=side_effect)
+
+    # When and Then
+    with pytest.raises(exceptions.FileSaveError):
+        test_django_file_domain_repository.save(
+            file_name=os.path.basename(test_const.INCORRECT_INPUT),
+            file=file_bytes,
+            location=str(tmp_path),
+        )
+
+
+def test_django_file_domain_repository_save_method_raises_file_save_error_on_suspicious_file_error(
+    tmp_path,
+    mocker: MockFixture,
+    test_django_file_domain_repository: ports.AbstractFileDomainRepository,
+):
+    # Given
+    with open(test_const.INCORRECT_INPUT, "rb") as file:
+        file_bytes = file.read()
+
+    side_effect = django_exceptions.SuspiciousFileOperation
+    mocker.patch.object(FileSystemStorage, "save", side_effect=side_effect)
+
+    # When and Then
+    with pytest.raises(exceptions.FileSaveError):
+        test_django_file_domain_repository.save(
+            file_name=os.path.basename(test_const.INCORRECT_INPUT),
+            file=file_bytes,
+            location=str(tmp_path),
+        )
+
+
+@pytest.mark.django_db
+def test_django_extract_domain_repository_create_method_creates_record(
+    test_django_extract_domain_repository: ports.AbstractExtractDomainRepository,
+):
+    # Given
+    extract_history_entity = value_objects.ExtractHistory(
+        "test_file.csv", "saved_file.csv", timestamp=datetime.now()
+    )
+    # When
+    test_django_extract_domain_repository.create(extract_history_entity)
+
+    # Then
+    assert (
+        models.ExtractHistory.objects.filter(
+            input_file_name=extract_history_entity.input_file_name
+        ).count()
+        == 1
+    )
+
+
+@pytest.mark.django_db
```

Comment:
This needs to be removed everywhere.

Add in tests/common/conftest.py:
```python
import pytest

from ..common.annotations import YieldFixture


@pytest.fixture(autouse=True)
def enable_db_access(transactional_db) -> YieldFixture:
    # Enable database access for all tests.
    yield
```

and then in tests/integarted/conftest.py
```python
from ..common.conftest import enable_db_access  # noqa: F401
```